tests/test_core.py

```diff
@@ -1318,3 +1359,32 @@ def test_objcinstance_retain_release(self):
         # Delete it and make sure that we don't segfault on garbage collection.
         del string
         gc.collect()
+
+    def test_objcinstance_dealloc(self):
+
+        class DeallocTester(NSObject):
+            attr0 = objc_property()
+            attr1 = objc_property(weak=True)
+
+            @objc_method
+            def dealloc(self):
+                self._did_dealloc = True
+
+        obj = DeallocTester.alloc().init()
+        obj.__dict__["_did_dealloc"] = False
+
+        attr0 = NSObject.alloc().init()
+        attr1 = NSObject.alloc().init()
+
+        obj.attr0 = attr0
+        obj.attr1 = attr1
+
+        assert attr0.retainCount() == 2
+        assert attr1.retainCount() == 1
```

Comment:
Only noticed this now - our tests use plain standard `unittest` and not PyTest, so these asserts should be written using `self.assertEqual` and not the `assert` keyword.

Reply:
Oops. Yes, now that you say that...